web/ui/src/Services/AppsController.js

```diff
@@ -292,34 +267,108 @@
                 $scope.templates.data = utils.mapToArr(templates);
             });
         }
-       
-        function fetchInternalServiceInstances(app) {
-            return resourcesFactory.v2.getInternalServiceInstances(app.id).then( data => {
-                app.instances = data;
+
+        function refreshApps() {
+            return fetchApps()
+                .then(fillOutApps)
+                .then(fetchServicesHealth)
+                .then(fetchInternalServicesHealth)
+                .then(() => touch());
+        }
+
+        function fetchApps() {
+            return $q.all([fetchInternalService(), fetchServices()])
+                .then(results => {
+                    $scope.apps = [];
+
+                    var isvc = results[0];
+                    $scope.apps.push(isvc);
+
+                    var services = results[1];
+                    services.forEach(s => {
+                        $scope.apps.push(s);
+                    });
+                });
+        }
+
+        function fetchChildEndpoints(app) {
+            return resourcesFactory.v2.getServiceChildPublicEndpoints(app.id).then(data => {
+                app.publicEndpoints = data;
```

Comment:
When the status is updated, there is a call to the fetchAllStates method on the Service object which sets the desiredState of the endpoint to its parent.